orc8r/gateway/python/magma/magmad/upgrade/upgrader.py

```diff
@@ -81,12 +83,19 @@ def start_upgrade_loop(magmad_service, upgrader):
 
     while True:
         logging.info('Checking for upgrade...')
-        try:
-            target_ver = _get_target_version(magmad_service.mconfig)
-            upgrader.perform_upgrade_if_necessary(target_ver)
-        except Exception:  # pylint: disable=broad-except
-            logging.exception(
-                'Error encountered while upgrading, will try again after delay',
+        target_ver = _get_target_version(magmad_service.mconfig)
+        if target_ver != DEFAULT_VERSION:
+            try:
+                upgrader.perform_upgrade_if_necessary(target_ver)
+            except Exception:  # pylint: disable=broad-except
+                logging.exception(
+                    'Error encountered while upgrading, '
+                    'will try again after delay',
+                )
```

Comment:
* If there is a retry here somewhere, would it be sensible to have a lower log level?
* Could the delay be specified more precisely here?
``` python
SLEEP_PERIOD = 120
...
yield from asyncio.sleep(SLEEP_PERIOD)
...
'will try again after %s', SLEEP_PERIOD
```
* Any reason for splitting the message in multiple lines?

Reply:
* It catches any exception so I assume it's an unexpected error and should have a high log level even if the retry happens.
* The sleep period is not a constant but we can put it into the log message. :+1:
* It's split to avoid having a very long line.